app/models/metric/statistic.rb

```diff
@@ -4,7 +4,14 @@ def self.calculate_stat_columns(obj, timestamp)
     return {} unless obj.respond_to?(:container_images)
 
     capture_interval = Metric::Helper.get_time_interval(obj, timestamp)
-    container_groups = ContainerGroup.where(:ems_id => obj.id).or(ContainerGroup.where(:old_ems_id => obj.id))
+
+    if obj.kind_of? ContainerProject
+      container_groups = ContainerGroup.where(:container_project_id => obj.id).or(ContainerGroup.where(:old_container_project_id => obj.id))
+    elsif obj.kind_of? ManageIQ::Providers::ContainerManager
+      container_groups = ContainerGroup.where(:ems_id => obj.id).or(ContainerGroup.where(:old_ems_id => obj.id))
+    else
+      container_groups = obj.container_groups
+    end
 
```

Comment:
the `if ... elsif ...` is not nice :-(

Another option is to add a `all_container_groups` method to `ContainerProject` and `ContainerManager`

In /app/models/container_project.rb b/app/models/container_project.rb
```
+  def all_container_groups
+    ContainerGroup.where(:container_project_id => id).or(ContainerGroup.where(:old_container_project_id => id))
+  end
```

In /app/models/manageiq/providers/container_manager.rb b/app/models/manageiq/providers/container_manager.rb

```
+    def all_container_groups
+      ContainerGroup.where(:ems_id => id).or(ContainerGroup.where(:old_ems_id => id))
+    end
```

And here do:
`return {} unless obj.respond_to?(:all_container_groups)`
and
`container_groups = obj.all_container_groups`